### PR TITLE
[branch-2.10] [fix] [broker] Fix isolated group not work problem.

### PR DESCRIPTION
**Pull request by horizonzy**
### Motivation
This is a cherry-pick commit for #21096.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/IsolatedBookieEnsemblePlacementPolicy.java [hunk 1]
```diff
@@ -187,17 +190,14 @@ private Set<BookieId> getBlacklistedBookiesWithIsolationGroups(int ensembleSize,
...
-                        bookieMappingCache.get(BookieRackAffinityMapping.BOOKIE_INFO_ROOT_PATH);
-
-                Optional<BookiesRackConfiguration> optRes = (future.isDone() && !future.isCompletedExceptionally())
-                        ? future.join() : Optional.empty();
-
-                if (!optRes.isPresent()) {
-                    return blacklistedBookies;
+                Optional<BookiesRackConfiguration> optional =
+                        bookieMappingCache.get(BookieRackAffinityMapping.BOOKIE_INFO_ROOT_PATH)
+                                .get(30, TimeUnit.SECONDS);
```
Comment (hangc0276) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/IsolatedBookieEnsemblePlacementPolicy.java:195]: Please do not hard code the timeout value

**Review by hangc0276 — COMMENTED** [review]
Please add a test to protect the logic

**Review by hangc0276 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by poorbarcode — APPROVED** [review]

**Comment by mattisonchao** [comment]
Please wait for this PR to fix some tests.

**Comment by mattisonchao** [comment]
Hi, @horizonzy 

You should rebase your branch to branch-2.10

**Pull request merged**